inductiva/utils/format_utils.py

```diff
@@ -40,55 +40,59 @@ def seconds_formatter(secs: float) -> str:
     return str(datetime.timedelta(seconds=round(secs)))
 
 
-def get_tabular_str(
-    rows,
-    columns,
-    default_col_space=15,
-    override_col_space=None,
-    formatters=None,
-) -> str:
-    """Converts a list of lists to a string table.
+def apply_formatters(table_data: dict, formatters: dict):
+    """Apply formatters to a list of lists of data (rows).
 
-    Temporary solution to display tables. `pandas` dependency to print tables
-    is overkill, but since we have `pandas` anyway for now, we can use it.
+    Args:
+        table_data (dict): Dictionary of column names and lists of data.
+        formatters (dict): Dictionary of column names and functions to
+            apply to that column's data.
     """
-    df = pd.DataFrame(rows, columns=columns)
-    formatters = formatters or {}
-
-    col_space = {col: default_col_space for col in columns}
-    col_space.update(override_col_space or {})
-
-    # replace None with np.nan so that pandas can format them as "n/a"
-    # by passing na_rep="n/a" to to_string()
-    df.fillna(np.nan, inplace=True)
-
-    return df.to_string(
-        index=False,
-        na_rep="n/a",
-        formatters=formatters,
-        col_space=col_space,
-    )
+    for column_name, formatter in formatters.items():
+        if column_name in table_data:
+            table_data[column_name] = [
+                formatter(x) for x in table_data[column_name]
+            ]
+
+    return table_data
+
+
+def get_tabular_str(tabular_data: Union[Mapping[str, Iterable[Any]],
+                                        Iterable[Iterable[Any]]],
+                    headers: list = (),
+                    formatters: dict = None) -> str:
+    """Converts the list rows to a string table.
+
+    Args:
+        rows (list): can be a list-of-lists (or another iterable of iterables),
+            a list of named tuples, a dictionary of iterables, an iterable of
+            dictionaries, an iterable of dataclasses (Python 3.7+)
+        columns (list): A list of column names.
+            Only needed if tabular_data is not a Mapping.
+        formatters (dict): A dictionary of column names and functions to apply
+            to the data in that column. The function should take a single
+            argument and return a string. The function will be applied to the
+            data in the column before printing. Defaults to None.
 
-
-def get_dataframe_str(dataframe,
-                      default_col_space=15,
-                      override_col_space=None,
-                      formatters=None):
-    """Converts a dataframe to a string table.
-
-    Temporary solution to display tables. `pandas` dependency to print tables
-    is overkill, but since we have `pandas` anyway for now, we can use it.
     """
+
     formatters = formatters or {}
 
-    col_space = {col: default_col_space for col in dataframe.columns}
-    col_space.update(override_col_space or {})
+    if not isinstance(tabular_data, Mapping):
```

Comment:
Is it just me or the `tabular_data` variable is not defined in the function body or arguments? Am I missing something?

Reply:
tabular_data is one of the arguments. On the doctoring I had rows by mistake.